packages/core/content-manager/admin/src/pages/EditView/components/FormInputs/BlocksInput/BlocksToolbar.tsx

```diff
@@ -538,32 +667,104 @@ const BlocksToolbar = () => {
+            </Flex>
+          </Toolbar.ToggleGroup>
+        </Flex>
+      ),
+      renderInMenu: () => [
+        <ListButton
+          block={blocks['list-unordered']}
+          format="unordered"
+          location="menu"
+          key="unordered"
+        />,
+        <ListButton
+          block={blocks['list-ordered']}
+          format="ordered"
+          location="menu"
+          key="ordered"
+        />,
+      ],
+    },
+  ] as Array<ObservedItem>;
+
+  const toolbarRef = React.useRef<HTMLElement>(null);
+  const [lastVisibleIndex, setLastVisibleIndex] = React.useState<number>(observedItems.length - 1);
+  const hasHiddenItems = lastVisibleIndex < observedItems.length - 1;
+
   return (
     <Toolbar.Root aria-disabled={disabled} asChild>
       <ToolbarWrapper gap={2} padding={2} width="100%">
         <BlocksDropdown />
         <Separator />
         <Toolbar.ToggleGroup type="multiple" asChild>
-          <Flex gap={1}>
-            {Object.entries(modifiers).map(([name, modifier]) => (
-              <ToolbarButton
-                key={name}
-                name={name}
-                icon={modifier.icon}
-                label={modifier.label}
-                isActive={modifier.checkIsActive(editor)}
-                handleClick={() => modifier.handleToggle(editor)}
-                disabled={isButtonDisabled}
-              />
-            ))}
-            <LinkButton disabled={isButtonDisabled} />
-          </Flex>
-        </Toolbar.ToggleGroup>
-        <Separator />
-        <Toolbar.ToggleGroup type="single" asChild>
-          <Flex gap={1}>
-            <ListButton block={blocks['list-unordered']} format="unordered" />
-            <ListButton block={blocks['list-ordered']} format="ordered" />
+          <Flex direction="row" gap={1} grow={1} overflow="hidden" ref={toolbarRef}>
+            {observedItems
+              .map((item, index) => (
+                <ObservedToolbarItem
+                  lastVisibleIndex={lastVisibleIndex}
+                  setLastVisibleIndex={setLastVisibleIndex}
+                  index={index}
+                  rootRef={toolbarRef}
+                  key={index}
+                >
+                  {item.renderInToolbar()}
+                </ObservedToolbarItem>
+              ))
+              /**
+               * Display the "more" menu in the right position among the items.
+               * We splice here after the map above to avoid messing with the indexes,
+               * since the ObservedToolbarItem component relies on them.
+               */
+              .toSpliced(
+                lastVisibleIndex + 1,
+                0,
+                <MoreMenu
+                  setLastVisibleIndex={setLastVisibleIndex}
+                  hasHiddenItems={hasHiddenItems}
+                  rootRef={toolbarRef}
+                >
+                  {observedItems.slice(lastVisibleIndex + 1).map((item) => item.renderInMenu())}
```

Comment:
Any value in storing `lastVisibleIndex + 1` in a variable with the other derived values since it's used twice?